### PR TITLE
Add Typescript testing docs

### LINKED ISSUE #472: Add page to document testing using testcontainers

### PR DESCRIPTION
**Pull request by jackkleeman**
FIxes #472

### CODE REVIEW

File: code_snippets/ts/src/develop/testing.test.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+    await state.setAll({ count: 1 });
+    // wont compile:
+    // state.setAll({ count: "a" });
+    // state.setAll({ foo: 1 });
+
+    expect(await state.getAll()).toStrictEqual({ count: 1 });
+    // wont compile:
+    // (await state.getAll()) satisfies { count: string };
+    // (await state.getAll()) satisfies { foo: number };
```
Comment (slinkydeveloper) [inline code_snippets/ts/src/develop/testing.test.ts:72]: Same

File: docs/develop/ts/testing.mdx [hunk 2]
```diff
@@ -0,0 +1,41 @@
+---
+sidebar_position: 12
```
Comment (slinkydeveloper) [inline docs/develop/ts/testing.mdx:2]: I honestly would put this page as last in the sidebar

**Review by slinkydeveloper — COMMENTED** [review]
Looks terrific. I would do just a few changes to make the page slimmer, i don't think we need to tell people too much about type safety, it's better to have the code example short

**Comment by eddumelendez** [comment]
Hi, I discovered the module and add it to the [Testcontainers Module Catalog](https://testcontainers.com/modules/restate/). If you have more to come please feel free to submit a PR updating [this file](https://github.com/testcontainers/community-module-registry/blob/main/modules/restate/index.md?plain=1)

**Comment by jackkleeman** [comment]
Whoa. Cool! Thanks @eddumelendez

**Pull request merged**